mc_providers/mediacloud.py

```diff
@@ -175,12 +176,18 @@ def _query(self, endpoint: str, params: Dict = None, method: str = 'GET'):
         if params and ('q' in params):
             params['q'] = sanitize_query(params['q'])
         endpoint_url = self.API_BASE_URL+endpoint
+        start = time.time()
         if method == 'GET':
             r = self._session.get(endpoint_url, params=params, timeout=self.TIMEOUT_SECS)
         elif method == 'POST':
             r = self._session.post(endpoint_url, json=params, timeout=self.TIMEOUT_SECS)
         else:
             raise RuntimeError("Unsupported method of '{}'".format(method))
+        duration = time.time() - start
+
+        if r.status_code == 504:
+            raise RuntimeError("API returned 504 Gateway Timeout after {} secs. Client timeout: {},"
```

Comment:
Maybe log the duration too?  "warning" might make it visible to sentry, not sure if that is a bug or feature (if it happens too often, it could enter the bug realm!)